query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -114,13 +115,20 @@ trait ExecPlan extends QueryCommand {
 
     // Step 2: Set up transformers and loop over all rangevectors, creating the result
     def step2(res: ExecResult) = res.schema.map { resSchema =>
+
       FiloSchedulers.assertThreadName(QuerySchedName)
       // It is possible a null schema is returned (due to no time series). In that case just return empty results
-      val resultTask = if (resSchema == ResultSchema.empty) {
+      val resSchemaForTransformer = if ((allTransformers.size > 0 &&
+        allTransformers.head.isInstanceOf[AbsentFunctionMapper])) {
+         ResultSchema(Seq(ColumnInfo("timestamp", ColumnType.LongColumn),
```

Comment:
This is not the right place for this.  The result schema returned should be returned in the AbsentFunctionMapper transformer itself, why can't it be done there?

Reply:
AbsentFunctionMapper is attached to the topmost plan- DistContExec.  Absent function returns range vectors when inner query returns no range vector. resSchema would be empty when no range vectors are returned by DistContExec, but we are supposed to return absent function result then. So I am creating ResultSchema for AbsentFunctionMapper

Reply:
I think there is still a way to not insert this code.   The AbsentFunctionMapper itself can change the schema that is output.